virtualizarr/manifests/manifest.py

```diff
@@ -54,6 +65,110 @@ def with_validation(
         return ChunkEntry(path=path, offset=offset, length=length)
 
 
+class ManifestStore(Store):
```

Comment:
I think this class should be defined in it's own file, and potentially even made public (developer) API.